src/pages/rt-fees-schedule-inw-map/Forms/FeeScheduleInwardsMapForm.js

```diff
@@ -0,0 +1,158 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { RemittanceOutwardsRepository } from 'src/repositories/RemittanceOutwardsRepository'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { RemittanceSettingsRepository } from 'src/repositories/RemittanceRepository'
+import { DataSets } from 'src/resources/DataSets'
+import { ControlContext } from 'src/providers/ControlContext'
+
+export default function FeeScheduleInwardsMapForm({ labels, maxAccess, recordId, record }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: RemittanceOutwardsRepository.FeeScheduleInwards.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: recordId,
+      corId: '',
+      dispersalMode: '',
+      scheduleId: ''
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      corId: yup.string().required(),
+      scheduleId: yup.string().required(),
+      dispersalMode: yup.string().required()
+    }),
+    onSubmit: async obj => {
+      const corId = formik.values.corId
+      const dispersalMode = formik.values.dispersalMode
+
+      await postRequest({
+        extension: RemittanceOutwardsRepository.FeeScheduleInwards.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!corId && !dispersalMode) {
+        toast.success(platformLabels.Added)
+      } else toast.success(platformLabels.Edited)
+
+      formik.setFieldValue(
+        'recordId',
+        String(obj.corId) + String(obj.dispersalMode)
+      )
+
+      invalidate()
+    }
+  })
+
+  const editMode = !!formik.values.recordId || !!recordId
```

Comment:
i guess const editMode = !!formik.values.recordId is enough 
unless we have special case here